src/make_data/banner_intervention/banner_select_reject_ignore.sql

```diff
@@ -0,0 +1,77 @@
+/* This query extracts the total number and proportion of sessions that select the
+   checker via the banner intervention, reject the checker via the banner
+   intervention, and that ignore the checker via the banner intervention.
+*/
+
+
+DECLARE start_date STRING DEFAULT "20210803";
+DECLARE end_date STRING DEFAULT "20210816";
+
+CREATE OR REPLACE TABLE `govuk-bigquery-analytics.banner_intervention.banner_select_reject_ignore` AS
+
+-- All sessions that are shown the checker via the banner (eventCategory =
+-- interventionBanner)
+
+WITH
+  sessions_banner AS (
+    SELECT DISTINCT
+      CONCAT(fullVisitorId, "-", visitId) AS sessionId,
+      hits.eventInfo.eventAction
+    FROM `govuk-bigquery-analytics.87773428.ga_sessions_*`
+    CROSS JOIN UNNEST(hits) AS hits
+    WHERE _TABLE_SUFFIX BETWEEN start_date AND end_date
+      AND hits.eventInfo.eventCategory = 'interventionBanner'
+  ),
+
+-- All distinct sessions that are shown the banner (eventCategory = interventionBanner;
+-- eventAction = interventionShown)
+
+  sessions_next_steps_shown AS (
```

Comment:
Many thanks for this suggestion. I think this does improve the readability but would appreciate any support in working out if this is more efficient. 

With this new script, I'd also be tempted to merge the first two subqueries into:


`
WITH sessions_flags AS ( 

     SELECT 
          CONCAT(fullVisitorId, "-", visitId) AS sessionId, 
          MAX(CASE WHEN hits.eventInfo.eventAction = 'interventionShown' THEN 1 ELSE 0 END) AS banner_shown,
          MAX(CASE WHEN hits.eventInfo.eventAction = 'interventionClicked' THEN 1 ELSE 0 END) AS banner_clicked, 
          MAX(CASE WHEN hits.eventInfo.eventAction = 'interventionDismissed' THEN 1 ELSE 0 END) AS banner_rejected 
      FROM `govuk-bigquery-analytics.87773428.ga_sessions_*` 
      CROSS JOIN UNNEST(hits) AS hits
      WHERE _TABLE_SUFFIX BETWEEN start_date AND end_date
      GROUP BY CONCAT(fullVisitorId, "-", visitId) 
), `

Do you think this worsens the readability, and should keep the two subqueries separate?

Reply:
I think that's a great idea! I would recommend keeping `WHERE hits.eventInfo.eventCategory = 'interventionBanner'` in, I think the main thing about efficiency in this case is limiting the amount of data we look at, because that's what charges are based on, but that doesn't change at all and I think this is the easiest to read.

Reply:
Thank you @suganya-s - that is super helpful. Have included `hits.eventInfo.eventCategory = 'interventionBanner'`

Refactored code in commit https://github.com/alphagov/govuk-entity-personalisation/pull/36/commits/7ee83cd766c596d53679c752f60ff770a243539a